packages/gatsby/cache-dir/static-entry.js

```diff
@@ -276,36 +238,35 @@ export default (pagePath, callback) => {
     )
   }
 
-  styles
-    .slice(0)
-    .reverse()
-    .forEach(style => {
-      // Add <link>s for styles that should be prefetched
-      // otherwise, inline as a <style> tag
-
-      if (style.rel === `prefetch`) {
-        headComponents.push(
-          <link
-            as="style"
-            rel={style.rel}
-            key={style.name}
-            href={`${__PATH_PREFIX__}/${style.name}`}
-          />
-        )
-      } else {
-        headComponents.unshift(
-          <style
-            data-href={`${__PATH_PREFIX__}/${style.name}`}
-            dangerouslySetInnerHTML={{
-              __html: fs.readFileSync(
-                join(process.cwd(), `public`, style.name),
-                `utf-8`
-              ),
-            }}
-          />
-        )
-      }
-    })
+  // Add <link>s for styles that should be prefetched
+  // otherwise, inline as a <style> tag
+  headComponents.push(
+    ...styles
```

Comment:
Overall, what you're doing here is pushing non-prefetch elements to the start of the array, then everything else, correct?

What if we re-wrote this slightly for clarity (or at least I _think_ it's clearer), e.g.

```js
const prefetchedAssets = styles
  .filter(s => s.rel === `prefetch`)
  .map(style => (
    <link
      as="style"
      rel={style.rel}
      key={style.file}
      href={`${__PATH_PREFIX__}/${style.file}`}
    />
  ))

const nonPrefetchedAssets = styles
    .filter(s => s.rel === `prefetch`)
    .map(style => (
      <style
        key={style.file}
        data-href={`${__PATH_PREFIX__}/${style.file}`}
        dangerouslySetInnerHTML={{
          __html: fs.readFileSync(
            join(process.cwd(), `public`, style.file), 'utf8'
          )
        }}
      />
    ))

headComponents.unshift(...nonPrefetchedAssets) // this clarifies things a bit, i.e. that these are first
headComponents.push(...prefetchedAssets)
```